s3path.py

```diff
@@ -1123,6 +1137,44 @@ def absolute(self):
         # We can't compute the absolute path from a relative one
         raise ValueError("Absolute path can't be determined for relative S3Path objects")
 
+    def get_presigned_url(self, expire_in: Union[timedelta, int] = 3600) -> str:
+        """
+        Returns a pre-signed url. Anyone with the url can make a GET request to get the file.
+        You can set an expiration date with the expire_in argument (integer or timedelta object).
+
+        Note that generating a presigned url may require more information or setup than to use other
+        S3Path functions. It's because it needs to know the exact aws region and use s3v4 as signature
+        version. Meaning you may have to do this:
+
+        ```python
+        import boto3
+        from botocore.config import Config
+        from s3path import S3Path, register_configuration_parameter
+
+        resource = boto3.resource(
+            "s3",
+            config=Config(signature_version="s3v4"),
+            region_name="the aws region name"
+        )
+        register_configuration_parameter(S3Path("/"), resource=resource)
+        ```
+
+        A simple example:
+        ```python
+        from s3path import S3Path
+        import requests
+
+        file = S3Path("/my-bucket/toto.txt")
+        file.write_text("hello world")
+
+        presigned_url = file.get_presigned_url()
+        print(requests.get(presigned_url).content)
+        b"hello world"
+        """
+        if isinstance(expire_in, timedelta):
```

Comment:
I think that we need to add here some validations
for example I would advice to add a call to _absolute_path_validation method.

Think if we have another validations that we need.
And also please add tests accordingly

Reply:
Done. Can you review again?